### PR TITLE
Docker and docker-compose configuration

### PR DESCRIPTION
**Pull request by mkurek**
- added Dockerfile for ralph
- added docker-compose (previously fig) configuration with mysql and nginx

connected to #1432

### CODE REVIEW

File: docker/ralph.conf [hunk 1]
```diff
@@ -0,0 +1,18 @@
+server {
+    listen 80;
+    server_name ralph;
+
+    location /docs {
+        alias /opt/ralph/docs/_build/html;
+    }
+
+#    location /static {
```
Comment (vi4m) [inline docker/ralph.conf:9]: Why is this commented out?

File: docker/upgrade.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+##!/bin/bash
```
Comment (vi4m) [inline docker/upgrade.sh:1]: to many ##

File: docker/upgrade.sh [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+set -e
+
+cd $RALPH_DIR
```
Comment (vi4m) [inline docker/upgrade.sh:4]: what will happen if RALPH_DIR is not set?

**Comment by vi4m** [comment]
Docker compose is not nearly production ready. We should point it out clearly in documentation.

File: docker/init.sh [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+set -e
+
+# wait for db init
+sleep 15
+
+ralph migrate
```
Comment (vi4m) [inline docker/init.sh:7]: always provide precise path for binary

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+web:
+  build: .
+  restart: always
+  environment:
+    - DJANGO_SETTINGS_MODULE=ralph.settings.settings_local
+  ports:
+    - "8000:8000"
+  volumes:
+    - docker/settings_local.py:/opt/ralph/src/ralph/settings/settings_local.py
```
Comment (vi4m) [inline docker-compose.yml:18]: docker configuration shoule be the same as in use in production. Why we decided to drop ~/.ralph/settings ?

File: docker/upgrade.sh (hunk 3 above)
Comment (mkurek) [inline docker/upgrade.sh:4]: RALPH_DIR will always be set (it's in Dockerfile).

File: docker-compose.yml (hunk 5 above)
Comment (mkurek) [inline docker-compose.yml:18]: I've added a comment that it's only temporary until ralph will handle local settings in other (better) way.

**Comment by mkurek** [comment]
@vi4m I've added info to documentation that it's not for production use right now (notice that docker-compose is one reason and second is using develop settings and packages - it's only for testing and fast deployment of test instances).

**Comment by vi4m** [comment]
Awesome.

**Comment by szok** [comment]
:ok:

**Comment by ar4s** [comment]
:ok:

**Comment by mkurek** [comment]
wait with merging it for now

**Pull request merged**